lib/run-checks.js

```diff
@@ -0,0 +1,75 @@
+// Description:
+//   Runs checks on repositories using github checksuite
+//
+//
+// Author:
+//   virneo
+
+module.exports = {
+  create,
+  update
+}
+
+async function create(context, name, status, output){
+  let headBranch
+  let headSha
+  
+  if (!output) {
+    output = {
+      title: 'Status-im Tests running',
+      summary: `Please be patient, we'll get you the result as soon as possible`
+    }
+  }
+  
+  if (context.payload.checksuite) {
+    // being called by check_run
+    headBranch = context.payload.checksuite.head_branch
```

Comment:
I see you've used checks API. As you already know checks API is generally used for automated tests and CI. This task list is manual and there is no automation in it. I think it should do more like WIP bot as mentioned in the issue description and add a pending status to PR until all tasks are marked as complete.

Reply:
Ok, Let me change that. This will also remove the need of upgrading probot to 6.2.8.

Reply:
Changed. Please review again. Verified working on my test repo. If you need an env and pem, I can share mine and you can review it on one of the PRs in my test repo.